consensus/src/epoch_manager.rs

```diff
@@ -1395,6 +1395,19 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
         fail_point!("consensus::process::any", |_| {
             Err(anyhow::anyhow!("Injected error in process_rpc_request"))
         });
+
+        match request.epoch() {
+            Some(epoch) if epoch != self.epoch() => {
+                monitor!(
+                    "process_different_epoch_rpc_request",
+                    self.process_different_epoch(epoch, peer_id)
+                )?;
+                return Ok(());
+            },
+            // TODO: differentiate between None and same epoch.
```

Comment:
Should None at least return early?

Reply:
No, BlockRetrieval doesn't have an epoch number for instance.